army.cpp

```diff
@@ -1271,16 +1333,22 @@ int Army::DoAnAttack(Battle * b, char const *special, int numAttacks, int attack
 	return ret;
 }
 
-void Army::Kill(int killed)
+void Army::Kill(int killed, int damage)
 {
 	Soldier *temp = soldiers[killed];
 
 	if (temp->amuletofi) return;
 
 	if (Globals->ARMY_ROUT == GameDefs::ARMY_ROUT_HITS_INDIVIDUAL)
 		hitsalive--;
-	temp->hits--;
-	temp->damage++;
+
+	int hitsLeft = temp->hits - damage;
```

Comment:
Probably instead of lines 1345-1351 it will be cleaner to use:

`    temp->damage += std::min(temp->hits, damage);`
`    temp->hits = std::max(0, temp->hits - damage);`